### PR TITLE
feat: dastewa/full trust apps support pass 0

### PR DESCRIPTION
**Pull request by dastewa**
Add support for full trust apps which requires that a resource (app registration) uri be passed to the getAuthToken call of the teams js library. The aim of this pr was to make minimal changes to the interfaces, and so this list of resources are considered optional. 

Additional context: In the case of full trust apps, the app registration uri placed in the Teams app manifest is not used by the Teams client for token procurement. Instead, full trust apps need the app registration uri to be passed to the teams client when the getAuthToken method is called. This is currently the only way SSO will work for full trust apps - without this change, the Teams client will **not** return a token to the TeamsFx library.

Related issue: https://github.com/OfficeDev/TeamsFx/issues/5978

### CODE REVIEW

**Review by SLdragon — CHANGES_REQUESTED** [review]
Looks good to me~
Tip: To pass Lint PR / check-format (pull_request), you should run `npm run setup` in the root folder of teamsfx, so that it will generate docs files for this change.

**Comment by SLdragon** [comment]
to pass the Lint PR / pr-title check, the PR title should be semantic, such as: feat: Dastewa/full trust apps support pass 0.

**Comment by dastewa** [comment]
@SLdragon - thank you so much for the feedback. I've made the updates to add the updated documentation as well as to update the title of the pr :).

**Comment by SLdragon** [comment]
E2E TEST: N/A

**Comment by SLdragon** [comment]
Lint PR / check-format (pull_request) still failed, seems missing this file: packages/sdk/review/teamsfx.api.md

**Comment by SLdragon** [comment]
Also try to pull latest code from dev to make codecov happy

**Review by SLdragon — APPROVED** [review]

File: packages/sdk/src/credential/teamsUserCredential.browser.ts [hunk 1]
```diff
@@ -185,7 +187,8 @@ export class TeamsUserCredential implements TokenCredential {
     options?: GetTokenOptions
   ): Promise<AccessToken | null> {
     validateScopesType(scopes);
-    const ssoToken = await this.getSSOToken();
+    const resources = (options as GetTeamsUserTokenOptions)?.resources;
```
Comment (tecton) [inline packages/sdk/src/credential/teamsUserCredential.browser.ts:190]: I guess the signature of `getToken` is unchanged in order to align with `TokenCredential` interface.
Do we need to add comment and example for this API feature? It looks like developers would need to pass the `resources` value in `options` parameter if they want to use full trust Teams apps.

Comment (dastewa) [reply packages/sdk/src/credential/teamsUserCredential.browser.ts:196]: @tecton that's a great call, I've updated the comments to include more information about this for full trust apps along with an example of how to use this. Is there anything else I need update for the "codecov/project" task to run?

Comment (tecton) [reply packages/sdk/src/credential/teamsUserCredential.browser.ts:196]: Maybe you can merge 'dev' branch again to trigger the task.

**Review by SLdragon — APPROVED** [review]

**Comment by SLdragon** [comment]
Seems codecov hangs, I will close this pr and reopen it to trigger it again.

**Pull request merged**